common/firestore/FirestoreClient.java

```diff
@@ -60,30 +60,47 @@ private String getCreateDocumentUrl(String user) {
     return getCreateDocumentUrl(user, null);
   }
 
+  private String getDocumentResponse(String path) throws IOException {
+    StringBuilder response = new StringBuilder();
+    URL url = new URL(getGetUrl(path));
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+    connection.setRequestMethod("GET");
+    connection.setRequestProperty("Authorization", "Bearer " + token);
+    try (BufferedReader reader =
+        new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
+      String line;
+      while ((line = reader.readLine()) != null) {
+        response.append(line);
+      }
+    }
+    if (connection.getResponseCode() != HTTP_OK) {
```

Comment:
@oferb shouldn't you check response code _before_ trying to read from it?

Reply:
The read `getInputStream()` actually does the call. Kind of odd but that's how it works.

https://stackoverflow.com/questions/16122999/java-urlconnection-when-do-i-need-to-use-the-connect-method

In the case of an error, it'll read the contents into `response` and then fail. I think it's not so bad to leave as is.